shell/utils/dynamic-content/types.d.ts

```diff
@@ -90,12 +90,37 @@ export type SupportInfo = {
   }
 };
 
+/**
+ * Call to action for an announcement
+ */
+export type CallToAction = {
+  action: string;
+  link: string;
+};
+
+/**
+ * Announcements to be shown in the notification center or on the home page
+ */
+export type Announcement = {
+  id: string; // Unique id for this announcement
+  title: string; // Title to be shown
+  message: string; // Message/Body for the announcement
+  target: string; // Where the announcement should be shown
```

Comment:
these would need to be above for docs to work.

```
/** Where the announcement should be shown */
target: string;
```

thinking this one in particular would be good to call out format (for notifications it's `notification/<level>`).

just something to help developers look at this and construct one without diving through code

Reply:
Added